beacon/blockchain/finalize_block.go

```diff
@@ -98,7 +99,22 @@ func (s *Service) FinalizeBlock(
 
 	// fetch and store the deposit for the block
 	blockNum := blk.GetBody().GetExecutionPayload().GetNumber()
-	s.depositFetcher(ctx, blockNum)
+
+	if isSyncing(req.Height, req.SyncingToHeight) {
+		s.logger.Info(
+			"Syncing in finalize_block, skipping deposit fetcher",
+			"height", req.Height, "syncing_to_height", req.SyncingToHeight,
+		)
+		// If we're syncing, we just use the consensus finalized block deposits as an optimization
+		// The DepositStore does not get included in AppHash calculation and hence introducing this will not result in AppHash.
+		if err = s.storageBackend.DepositStore().EnqueueDeposits(ctx, blk.GetBody().GetDeposits()); err != nil {
+			return nil, fmt.Errorf("failed to enqueue deposits: %w", err)
+		}
+	} else {
+		// If we're NOT syncing we want to run the deposit fetcher as we want to have our own view
+		// of the deposits which we can validate in ProcessProposal.
+		s.depositFetcher(ctx, blockNum)
```

Comment:
super nit: can you just remove `blockNum` defn from line 100
```suggestion
		s.depositFetcher(ctx, blk.GetBody().GetExecutionPayload().GetNumber())
```